plugins/python/redisgears_python.c

```diff
@@ -1746,6 +1751,550 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+        if(!PyUnicode_Check(output)){
+            PyErr_SetString(GearsError, "Output name must be a string");
+            goto cleanup;
+        }
+        const char* outputStr = PyUnicode_AsUTF8AndSize(output, NULL);
+        RedisAI_DAGRunOpAddOutput(modelRunOp, outputStr);
+    }
+    if (RedisAI_DAGAddRunOp(pyDag->dag, modelRunOp, err) != REDISMODULE_OK) {
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        goto cleanup;
+    }
+    RedisAI_FreeError(err);
+    Py_INCREF(self);
+    return self;
+
+    cleanup:
+    RedisAI_FreeError(err);
+    RedisAI_DAGRunOpFree(modelRunOp);
+    return NULL;
+}
+
+static PyObject* appendScriptRunOp(PyObject *self, PyObject *args, PyObject *kargs) {
+    verifyRedisAILoaded();
+    if (RedisAI_DAGCreateScriptRunOp == NULL) {
+        PyErr_SetString(GearsError, "DAG run is not supported in RedisAI version");
+        return NULL;
+    }
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+
+    // Load script from keyspace and create SCRIPTRUN op.
+    PyObject* scriptName = GearsPyDict_GetItemString(kargs, "name");
+    if (!scriptName) {
+        PyErr_SetString(GearsError, "Script key name was not given");
+        return NULL;
+    }
+    if(!PyUnicode_Check(scriptName)){
+        PyErr_SetString(GearsError, "Script name argument must be a string");
+        return NULL;
+    }
+    PyObject* funcName = GearsPyDict_GetItemString(kargs, "func");
+    if (!funcName) {
+        PyErr_SetString(GearsError, "Function name was not given");
+        return NULL;
+    }
+    if(!PyUnicode_Check(funcName)){
+        PyErr_SetString(GearsError, "Function name argument must be a string");
+        return NULL;
+    }
+    const char* scriptNameStr = PyUnicode_AsUTF8AndSize(scriptName, NULL);
+    const char* functionNameStr = PyUnicode_AsUTF8AndSize(funcName, NULL);
+    RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
+    RedisGears_LockHanlderAcquire(ctx);
+    RedisModuleString* keyRedisStr = RedisModule_CreateString(ctx, scriptNameStr, strlen(scriptNameStr));
+    RedisModuleKey *key;
+    RAI_Error *err;
+    RedisAI_InitError(&err);
+    RAI_Script *script;
+    if(RedisAI_GetScriptFromKeyspace(ctx, keyRedisStr, &key, &script, REDISMODULE_READ, err) != REDISMODULE_OK) {
+        RedisModule_FreeString(ctx, keyRedisStr);
+        RedisGears_LockHanlderRelease(ctx);
+        RedisModule_FreeThreadSafeContext(ctx);
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        return NULL;
+    }
+
+    RedisModule_FreeString(ctx, keyRedisStr);
+    RedisGears_LockHanlderRelease(ctx);
+    RedisModule_FreeThreadSafeContext(ctx);
```

Comment:
move this into a function, for cleaner code

Reply:
done